extract/src/main/java/org/codice/imaging/nitf/extract/ImageSegmentHandler.java

```diff
@@ -0,0 +1,225 @@
+          .put(
+              "ICOM2",
+              segment ->
+                  segment.getImageComments().size() > 1 ? segment.getImageComments().get(1) : "")
+          .put(
+              "ICOM3",
+              segment ->
+                  segment.getImageComments().size() > 2 ? segment.getImageComments().get(2) : "")
+          .put(
+              "ICOM4",
+              segment ->
+                  segment.getImageComments().size() > 3 ? segment.getImageComments().get(3) : "")
+          .put(
+              "ICOM5",
+              segment ->
+                  segment.getImageComments().size() > 4 ? segment.getImageComments().get(4) : "")
+          .put(
+              "ICOM6",
+              segment ->
+                  segment.getImageComments().size() > 5 ? segment.getImageComments().get(5) : "")
+          .put(
+              "ICOM7",
+              segment ->
+                  segment.getImageComments().size() > 6 ? segment.getImageComments().get(6) : "")
+          .put(
+              "ICOM8",
+              segment ->
+                  segment.getImageComments().size() > 7 ? segment.getImageComments().get(7) : "")
+          .put(
+              "ICOM9",
+              segment ->
+                  segment.getImageComments().size() > 8 ? segment.getImageComments().get(8) : "")
+          .put("NBANDS", segment -> Integer.toString(segment.getNumBands()))
+          .put("IMODE", segment -> segment.getImageMode().getTextEquivalent())
+          .put("NBPR", segment -> Integer.toString(segment.getNumberOfBlocksPerRow()))
+          .put("NBPC", segment -> Integer.toString(segment.getNumberOfBlocksPerColumn()))
+          .put("NPPBH", segment -> Long.toString(segment.getNumberOfPixelsPerBlockHorizontal()))
+          .put("NPPBV", segment -> Long.toString(segment.getNumberOfPixelsPerBlockVertical()))
+          .put("NBPP", segment -> Integer.toString(segment.getNumberOfBitsPerPixelPerBand()))
+          .put("IDLVL", segment -> Integer.toString(segment.getImageDisplayLevel()))
+          .put("IALVL", segment -> Integer.toString(segment.getAttachmentLevel()))
+          .put(
+              "ILOC",
+              segment -> segment.getImageLocationRow() + "," + segment.getImageLocationColumn())
+          .put("IMAG", segment -> Double.toString(segment.getImageMagnificationAsDouble()))
+          .build();
+
+  ImageSegmentHandler(BiConsumer<ImageSegment, Map<String, String>> finalProcessing) {
+    super("image", MAPPING, finalProcessing);
+  }
+
+  private static String getTargetId(ImageSegment imageSegment) {
+    try {
+      final TargetId targetId = imageSegment.getImageTargetId();
+      if (targetId != null) {
+        final String targetIdValue = targetId.textValue();
+        if (StringUtils.isNotBlank(targetIdValue)) {
+          return targetIdValue;
+        }
+      }
+    } catch (NitfFormatException e) {
+      LOGGER.debug(
```

Comment:
✏️ Include the exception in the log statement

Reply:
fixed